### ISSUE TITLE
Login page with CSS / JS / PHP needs to be created

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/RadioLogger#9]
I intended on using a modified version of the code here for the login page which will appear as soon as the user starts the app.  Splash page like and stuff.

Please see https://www.w3schools.com/howto/tryit.asp?filename=tryhow_css_login_form

I tried to get this css working but it did not work well at all when I ran the localhost preview in Firefox so that it looks more or less like the example.  I tend to like transparent login pages like the one in this codepen.

https://codepen.io/Dru-Go/pen/jONobQB

I would like to use Axios for security and authentication when it comes to the PERN stack app rather than roll our own services.
See: https://github.com/axios/axios

The user should be able to create his her own app avatar for display when logging in.

To log out/ sign out, the user should be able to go to a logout toggle somewhere.

There should also be functionality to include "Forgot username" and "Forgot Password"

**Comment by idormain** [comment]
So we’re using nodejs now? Not PHP or both? I think it’s a mistake to use so many different technologies such as Axios, etc. we should be building a MVP first in the a technology stack before we include and bind together other pieces. Just my opinion.

**Comment by gbowne1** [comment]
@idormain 

So here goes, I forgot that the stale issue bot closed these issues using on RadioLogger so I re-opened them manually  @today.

I had also created a second repository because after some discussion with a few people, I felt that I was trying to build too much too fast.

The files in this particular version of the repository do not work.  Mostly because I was working backwards.

`radio-logger` over at [radio-logger](https://www.github.com/gbowne1/radio-logger/) does actually load into the browser.   I only added the Material UI library.   

SO, the two needs to be merged so there's only one RadioLogger repository.. I'm totally okay with keeping either one going forwards and removing the other.  I just would assume that we keep the one that currently works.  There may be some useful files in this repository.

So the base stack is PostgreSQL ExpressJS, REACT, NodeJS and was created with npm's Create REACT App (cra) instead of inside Visual Studio MVC REACT project mostly because I no longer use Windows to develop things.  I know you've said you hadn't time to learn Postgres, etc. so I left this project where it was not being able to find anyone to help out.

The MySQL/MariaDB/MongoDB, Express, Angular, Node stack would be okay too.

I have the radio-logger PostgreSQL DB hosted on Heroku as well as created a local pgAmin server for RadioLogger.  I had intended on locally hosting the App and its database, unless it gets huge.  However, since the move, I've created a MySQL/Maria/phpMyAdmin version.  This is the repository linked to the Heroku instead of the `radio-logger` version.  I'd have to go re-set up the PostgreSQL/pgAdmin version.

We're  gonna have to connect several services through API's to get anything else working.  Including this one one here:  https://wiki.radioreference.com/index.php/API

I was not sure at the time I created this project how we should handle Login.  There are several methods it seems.  I was advised by a number of REACT developers not to roll our own authentication.  

All in all this project needs a lot of cleanup.  Thoughts?

**Comment by idormain** [comment]
Hi Greg

Ok, let’s look at it, since you’re call for help with this project, how many qualified people have put their hands up for this technology stack? Out of these people, how has the progress been?

The website you’re after is not enterprise grade, therefore, using technologies like REACT aren’t needed because the site itself doesn’t need reusable components yet. 

We’re not talking a thousand page site in reality we’re talking say five to ten pages. Now, I understand reusable components but are five to ten pages going to have sufficient use in design that each page requires one or more reusable components?

In my opinion, developing our own authentication means we’re not reliant on other projects being maintained and there potentially be less maintenance of our own site due to self dependence. 

By using technologies like MySQL, etc means you can tap into the hobbyists market with a better chance of getting the help we need.  REACT and NodeJS are not really technologies hobbyists use daily due to their particular use cases. 

In my opinion, we should use; MySQL, PHP, JavaScript, CSS and HTML while using say JQuery. 

I think these are all we need for rapid application development and prototyping. 

Technology stacks are specialised for a reason and can have a steep learning curve unless your people use them daily. 

These are my thoughts and opinions. Your thoughts?

Cheers

Ivan

**Comment by gbowne1** [comment]
> Ok, let’s look at it, since you’re call for help with this project, how many qualified people have put their hands up for this technology stack? Out of these people, how has the progress been?

I have not aggressively had the time to market the development of this app becuase of my move and business needs.  So 10 or less and I've been the only one other than you, making it.  I also realize I have not really planned this project that well either.  But time to get back on track.

> The website you’re after is not enterprise grade, therefore, using technologies like REACT aren’t needed because the site itself doesn’t need reusable components yet.

No, you're right it's not currently at that level yet.  There is not reason that it couldn't be in the future.  

>We’re not talking a thousand page site in reality we’re talking say five to ten pages. Now, I understand reusable components but are five to ten pages going to have sufficient use in design that each page requires one or more reusable components?

Yeah, if you clone and load the current working app from the other repository [radio-logger](https://www.github.com/gbowne1/radio-logger) with `npm install`  and then run `npm start` then you can see that it only needs 20 or less pages all in, and most of them would be forms of some type other than some method of entering and/or displaying data and interacting with the database.

>In my opinion, developing our own authentication means we’re not reliant on other projects being maintained and there potentially be less maintenance of our own site due to self dependence.

True.  REACT and it's technologies have only been around for 10 or so years.  The others have been around longer.

>By using technologies like MySQL, etc means you can tap into the hobbyists market with a better chance of getting the help we need.  REACT and NodeJS are not really technologies hobbyists use daily due to their particular use cases.

True.  But I have seen plenty of apps that use them.  That does not mean that they were done very well.  Which I think this one could be.

>In my opinion, we should use; MySQL, PHP, JavaScript, CSS and HTML while using say JQuery.

True. This would be fine as a MVP.  And I already know these fairly well, only having needed to come up to speed with recent changes.  [if you've read my profile here [Profile](https://www.github.com/gbowne1/gbowne1) and you can see all of the `learning_<thing here> ` repos. I have

>I think these are all we need for rapid application development and prototyping.

Yes.  I am still learning PostgreSQL, Express, REACT, Node... and I think I now have a understanding of Node and Express but not so much REACT.  Until I've completed my learning journey there it might be best to get a simpler technology working.  

These are my thoughts and opinions. Your thoughts?

Thanks for your thoughts and opinions, Iv.

**Comment by idormain** [comment]
No problems Greg I’ll be developing with you in Linux now that I have a dedicated computer running it. I’ll have some time spare this week so let me know what you need done. 

I’ll look at the GitHub pages as well. 

Cheers

Ivan

**Comment by gbowne1** [comment]
We can leave `radio-logger` repository as it is.  It works.  Just could leave be for the time being or/if we ever get to the point where PostgreSQL, Express, REACT, Node ever become a option.   

But.. continue forwards with this repository and clean it up // remove all the unnecessary files..  I have these particular files backed up for use in radio-logger at that point.  

I think at the very, very, very minimum we need a .github and .github/Workflows folder as well as the .vs and .vscode folder and a base index.html and database(s).  I already have the MySQL/MariaDB/phpMyAdmin setup running with a db with few tables.

Yeah, have a look.  Oh and a side note, I updated both of the TAOT repos and tested the `taotnpwaft/website/index.html` locally with localhost.,  Pics in #webprogramming in Discord.  Also tested the open/close of the Login modal button.

**Comment by gbowne1** [comment]
[Note: gonna work on this 12/06/2021-12/09/2021]

**Comment by idormain** [comment]
Hiya 

Can you tell me what GitHub project the radiologger is I think I’ve already got it linked but just want to double check the name. Also the database? I’ve now got my Zorin Linux setup with both JetBrains DataGrip and PHPStorm. Ready to clone and start!

**Comment by gbowne1** [comment]
@idormain

Yeah the working one is https://www.github.com/gbowne1/radio-logger/  but it's REACT.  Are we sticking with that?

The one in gbowne1/RadioLogger/ doesn't work but was the first iteration.  [and could be replaced]

I can give you the database details of the MySQL/MariaDB/phpMyAdmin one I created not that long ago.

**Comment by idormain** [comment]
I don’t believe we need to use REACT? Yes, send me the database details.

**Comment by gbowne1** [comment]
Ok.  I'll Discord you the IP address.  I have it running locally.  I'm still in the process of creating the machine it will eventually live on.  Port 3306

**Comment by idormain** [comment]
Great. I’ll finish setting up in the more.

**Comment by gbowne1** [comment]
Yeah I'll be on to chat for the next 3-4 hours.  let me know if you want a custom user::password 

I have snapshots of what the REACT UI looked like.. but it sure could be way better.
